server/command.go

```diff
@@ -327,6 +329,61 @@ func (p *Plugin) runPopCommand(args []string, extra *model.CommandArgs) (bool, e
 	return false, nil
 }
 
+func (p *Plugin) runRemoveCommand(args []string, extra *model.CommandArgs) (bool, error) {
+	message := strings.Join(args, " ")
```

Comment:
Having to write down the whole message seems bad UX, but 0/5.

Reply:
True. Maybe we can only take the number of Todo in the remove cmd and delete that specific todo. In that case, I think we should change the list of todos from a bullet list to a numbered list. What do you think about this?

Reply:
Numbered list is "dangerous". I get my list: `1. todo A, 2. todo B, 3. todo C`. Now I say... delete number 2 (`todo B`). But while I am writing that, another user deletes the first one `todo A`. And I end up deleting the "new second one", `todo C`.

Something I did for the badges plugin was to use the autocomplete to add the badge id (in this case the todo id). May be an option, but 0/5 on how good. I don't like much the way it is done in the badges plugin neither :sweat_smile:

A more elaborate option (and maybe not good neither) is that `/todo delete` opens an interactive dialog with all the todos, and you mark with a checkmark all of the todos you want to delete.

@asaadmahmood Since you were doing some changes on the todo plugin some time ago, not sure if you have any thoughts on the UX for this particular part (slash command for todo deletion).

Reply:
@asaadmahmood any suggestion here?

Reply:
@thoratvinod @larkox I think we should have an autocomplete with the todo list.
So when the user types `/todo remove ` an autocomplete opens up, with the todo list (name+description), and then the user can select the one he wants to remove.

When he selects it, the selection can turn into an `id` for simplicity.

Reply:
Thanks for the suggestion @asaadmahmood. However, I'm not sure how we can implement this in the code.

After reading the code, I've noticed a few things. Please correct me if I'm wrong:
1. The `OnActivate` method will run only once when the plugin is activated.
2. The `ExecuteCommand` method will run every time a command is entered.
3. The `OnDeactivate` method will run only once when the plugin is deactivated.

We register all plugin commands in the `OnActivate` method, which is executed only once after the plugin is activated. If we want to add autocomplete for the delete command, we would need to fetch the latest todo list, which I believe is not possible.

@mickmister @larkox please let me know if I am missing something or if you have any way to enable autocomplete for delete.

Reply:
Take a look to https://github.com/larkox/mattermost-plugin-badges

The idea is to register the autocomplete as something dynamic. Hopefully reading the code in there is enough, but if you need more guidance let me know :smile:

Reply:
Have made changes, also updated the screenshots.